### PR TITLE
Jwks file

### PR DESCRIPTION
**Pull request by mortenmj**
This makes it possible to periodically reload JWKS content from a file. We do this by replacing the current jwks_inline field in the JWT proto with a oneoff that can be either inline content, or a file path + a refresh interval.

In the case that this is a file path, we create a ForwardingSignatureValidator, which forwards requests to ValidateSignature to some internal SignatureValidator it holds a pointer to. We then periodically read the JWKS file content, and update the internal SignatureValidator if required.

### CODE REVIEW

File: pkg/jwt/configuration.go [hunk 1]
```diff
@@ -23,17 +26,24 @@ import (
...
+		var jwks jose.JSONWebKeySet
+		if err := json.Unmarshal(jwksJSON, &jwks); err != nil {
+			return nil, util.StatusWrapWithCode(err, codes.InvalidArgument, "Failed to unmarshal JSON Web Key Set")
+		}
+		signatureValidator, err = NewSignatureValidatorFromJSONWebKeySet(&jwks)
+		if err != nil {
+			return nil, err
+		}
+	case *configuration.AuthorizationHeaderParserConfiguration_JwksFile_:
+		signatureValidator = NewSignatureValidatorFromJSONWebKeySetFile(key.JwksFile.FilePath, key.JwksFile.RefreshInterval.AsDuration())
```
Comment (tokongs) [inline pkg/jwt/configuration.go]: Maybe add a default case that returns an error

Comment (mortenmj) [reply pkg/jwt/configuration.go]: Done

File: pkg/jwt/configuration.go [hunk 2]
```diff
@@ -102,3 +112,39 @@ func NewSignatureValidatorFromJSONWebKeySet(jwks *jose.JSONWebKeySet) (Signature
...
+func NewSignatureValidatorFromJSONWebKeySetFile(path string, refreshInterval time.Duration) SignatureValidator {
+	// Hmm. I don't want us to first read the file once, to initialize this, and then read the file periodically to update it.
+	// However, I also don't want us to initialize this in a state that isn't ready for use.
+	validator := NewForwardingSignatureValidator(nil)
+
+	// TODO: Run this as part of the program.Group, so that it gets
+	// cleaned up upon shutdown.
+	go func() {
+		t := time.NewTicker(refreshInterval)
+		for {
```
Comment (tokongs) [inline pkg/jwt/configuration.go:139]: nit: `for _ = range t.C` might be a bit more idiomatic

Comment (mortenmj) [reply pkg/jwt/configuration.go:139]: Done

File: pkg/jwt/forwarding_signature_validator.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	"sync/atomic"
+)
+
+type forwardingSignatureValidator struct {
+	validator atomic.Pointer[SignatureValidator]
+}
+
+// NewForwardingSignatureValidator creates a SignatureValidator that simply forwards
+// requests to another SignatureValidator.
+func NewForwardingSignatureValidator(validator SignatureValidator) SignatureValidator {
```
Comment (tokongs) [inline pkg/jwt/forwarding_signature_validator.go]: Any reason to not return the concrete struct here? Common go pattern is to "Accept interfaces, return structs"

Comment (mortenmj) [reply pkg/jwt/forwarding_signature_validator.go]: It's the pattern they've adopted, so I'm forced to follow that I think

**Pull request closed without merging**